src/core/Form/Label/Label.tsx

```diff
@@ -40,42 +60,70 @@ const labelTextClassNames = {
 const StyledLabel = styled(
   ({
     className,
+    style,
+    contentClassName,
     theme,
     labelMode = 'visible',
-    labelSpanProps = { className: undefined },
+    wrapperProps,
     children,
     asProp = 'label',
     optionalText,
+    tooltipComponent: tooltipComponentProp,
     ...passProps
-  }: LabelProps & SuomifiThemeProp) => (
-    <HtmlDiv
-      {...(asProp ? { as: asProp } : {})}
-      className={classnames(className, baseClassName)}
-      {...passProps}
-    >
-      {labelMode === 'hidden' ? (
-        <VisuallyHidden>
-          {children}
-          {optionalText && `(${optionalText})`}
-        </VisuallyHidden>
-      ) : (
-        <HtmlSpan
-          {...labelSpanProps}
-          className={classnames(
-            labelTextClassNames.labelSpan,
-            labelSpanProps.className,
-          )}
-        >
-          {children}
-          {optionalText && (
-            <HtmlSpan className={labelTextClassNames.optionalText}>
-              {` (${optionalText})`}
+  }: LabelProps & SuomifiThemeProp) => {
+    const [wrapperRef, setWrapperRef] = useState<HTMLDivElement | null>(null);
+
+    function getTooltipComponent(
+      tooltipComponent: ReactElement | undefined,
+    ): ReactNode {
+      if (isValidElement(tooltipComponent)) {
+        return cloneElement(tooltipComponent, {
+          anchorElement: wrapperRef,
+          // trick to force tooltip to rerender every time when label changes.
+          key: Date.now(),
+        });
+      }
+      return null;
+    }
+
+    return (
+      <HtmlDivWithRef
+        className={classnames(className, baseClassName)}
+        {...wrapperProps}
+        forwardedRef={(ref: SetStateAction<HTMLDivElement | null>) =>
+          setWrapperRef(ref)
+        }
+      >
+        {labelMode === 'hidden' ? (
+          <VisuallyHidden>
+            {children}
+            {optionalText && `(${optionalText})`}
+          </VisuallyHidden>
+        ) : (
+          <>
+            <HtmlSpan
+              as={asProp}
+              {...(style ? { style } : {})}
+              {...passProps}
+              className={classnames(
+                labelTextClassNames.labelSpan,
+                contentClassName,
+              )}
+            >
+              {children}
+              {optionalText && (
+                <HtmlSpan className={labelTextClassNames.optionalText}>
+                  {` (${optionalText})`}
+                </HtmlSpan>
+              )}
             </HtmlSpan>
-          )}
-        </HtmlSpan>
-      )}
-    </HtmlDiv>
-  ),
+            {!!tooltipComponentProp &&
+              getTooltipComponent(tooltipComponentProp)}
```

Comment:
Quickly tested to just use like the following, **IF** using the `typeof Tooltip` for the prop as in my other comment.
```tsx
{!!tooltipComponentProp && tooltipComponentProp}
```

So far it seems to work without the complex logic for checking validity and need for cloning. But needs more testing to see if hiccups will occur.

Reply:
There are cases where tooltips will not be correctly updated if tooltip anchors (text length, container size or what ever the anchor element is, might be inline, block, flex etc) gets updated, so there needs to be a way to force a rerender. Key was used to achieve this and reposition the css arrow. If there is an alternative way that works 100% of the time, the additional logic can and should be removed.

Reply:
Leaving this unchanged.